dcicutils/ecr_utils.py

```diff
@@ -0,0 +1,68 @@
+import boto3
+import base64
+from .misc_utils import PRINT
+
+
+# Defines the standard Docker image tags for CGAP ECR
+# Latest and stable - something like this should be reasonable?
+CGAP_ECR_LAYOUT = {
+    'latest-wsgi': 'Latest version of WSGI Application',
+    'latest-indexer': 'Latest version of Indexer Application',
+    'latest-ingester': 'Latest version of the Ingester Application',
+    'stable-wsgi': 'Stable version of WSGI Application',
+    'stable-indexer': 'Stable version of Indexer Application',
+    'stable-ingester': 'Stable version of the Ingester Application'
+}
+
+
+class ECRUtils(object):
+    """ Utility class for interacting with ECR.
+        Initialized with an env name, from which a repo URL is resolved.
+
+        NOTE 1: unlike s3Utils, the resolved resources (repository) will NOT be
+        created if it does not already exist.
+        NOTE 2: the (already created) ECR repository must have the env_name in the URI
+    """
+
+    def __init__(self, *, env_name='cgap-mastertest', local_repository='cgap-wsgi'):
+        """ Creates an ECR client on startup """
+        self.env = env_name
+        self.local_repository = local_repository
+        self.is_cgap = 'cgap' in env_name
+        self.client = boto3.client('ecr', region='us-east-1')  # XXX: constant?
+        self.url = None  # set by calling the below method
+
+    def resolve_repository_uri(self, url=None):
+        if not self.url or url:
+            PRINT('NOTE: Calling out to ECR')
+            try:
+                resp = self.client.describe_repositories()
+                for repo in resp.get('repositories', []):
+                    if self.env in repo['repositoryUri']:
```

Comment:
It was not obvious to me at first that this is a text search. I suspect it will match `fourfront-cgapdev` while searching for `fourfront-cgap`. I would like to see a real URL parser applied here or some more robust test. I worry this could cause some serious dyslexia between environments if not careful.

I regard this comment as blocking, though there are a number of ways to resolve it.

Reply:
Fixed with `if repo['repositoryUri'].endswith(self.env)`